### PR TITLE
Marlin plugin

### PR DESCRIPTION
**Pull request by sithlord48**
Create a marlin plugin 
Better detection of Firmware names by checking with less junk

### CODE REVIEW

File: src/atcore.cpp [hunk 1]
```diff
@@ -101,9 +101,12 @@ void AtCore::findFirmware(const QByteArray &message)
         if (file.startsWith(QStringLiteral("lib"))) {
             file = file.remove(QStringLiteral("lib"));
         }
+        QByteArray fwName = message;
+        fwName.remove(0, 14); //remove FIRMWARE_NAME:
```
Comment (patrickelectric) [inline src/atcore.cpp]: This form isn't robust. Maybe a combination of contains, split and other things.

File: src/atcore.cpp [hunk 2]
```diff
@@ -101,9 +101,12 @@ void AtCore::findFirmware(const QByteArray &message)
...
+        fwName.resize(fwName.indexOf(' '));
 
-        if (!message.contains(file.toLocal8Bit())) {
-            qDebug() << "But it's not the plugin for this firmware." << message;
+        if (!fwName.toLower().contains(file.toLocal8Bit())) {
```
Comment (patrickelectric) [inline src/atcore.cpp]: Before that, use toLower and simplifie in message and file

**Review by patrickelectric — CHANGES_REQUESTED** [review]

File: src/atcore.cpp (hunk 2 above)
Comment (sithlord48) [reply src/atcore.cpp]: :heavy_check_mark:  `fwName = fwName.toLower().simplified();`

File: src/atcore.cpp (hunk 1 above)
Comment (sithlord48) [reply src/atcore.cpp]: :heavy_check_mark: `fwName.split(':').at(1)`

File: src/atcore.cpp (hunk 2 above)
Comment (patrickelectric) [reply src/atcore.cpp]: @sithlord48  and **file** variable.

Comment (sithlord48) [reply src/atcore.cpp]: :heavy_check_mark:  `file = file.toLower().simplified();`

**Review by patrickelectric — APPROVED** [review]
great work @sithlord48  !

**Pull request merged**